lib/periodicChecks.js

```diff
@@ -0,0 +1,45 @@
+const assignUnassignedReviewers = async (pullRequest, context) => {
+  const assignees = pullRequest.assignees.map(user => user.login);
+  const reviewers = pullRequest.requested_reviewers.map(user => user.login);
+
+  // If the PR author is assigned, then don't assign other reviewers.
+  if (assignees.includes(pullRequest.user.login)) {
```

Comment:
What if any reviewer has requested changes on the PR but missed assigning the PR author?

Reply:
(Just a thought, maybe we can handle this without scheduler i.e, once a reviewer submits PR review with "request-changes" then the bot will unassign the reviewer and assign the author.)

Reply:
There is a feature in the bot sheet that assigns the user automatically when a reviewer request changes on a PR.
We can't check from the PR that a reviewer requested changes at this point (from the PR data).

Reply:
> (Just a thought, maybe we can handle this without scheduler i.e, once a reviewer submits PR review with "request-changes" then the bot will unassign the reviewer and assign the author.)

Yes @DubeySandeep, that is part of the oppiabot features.

Reply:
Yes, let us have that check in this PR as well (we will need an upgrade later but it won't be difficult to upgrade - on the other hand not having this feature would definitely be a problem, since the periodic check is somewhat incomplete without it).

Reply:
Done.